types/time.go

```diff
@@ -1224,12 +1218,12 @@ func parseTime(sc *stmtctx.StatementContext, str string, tp byte, fsp int, isFlo
 
 // ParseDatetime is a helper function wrapping ParseTime with datetime type and default fsp.
 func ParseDatetime(sc *stmtctx.StatementContext, str string) (Time, error) {
-	return ParseTime(sc, str, mysql.TypeDatetime, DefaultFsp)
+	return ParseTime(sc, str, mysql.TypeDatetime, GetFsp(str))
```

Comment:
Is this line the whole change of this PR?

Reply:
And this cast branch:
https://github.com/pingcap/tidb/pull/6451/files#diff-88da33c1e3eaab083f98c6e9e260373aR1012